sdk/core/System.ServiceModel.Rest/src/Temporary/ClientUtilities.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.ServiceModel.Rest.Shared
+{
+    /// <summary>
+    /// TBD.
+    /// </summary>
+    public class ClientUtilities
+    {
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="value"></param>
+        /// <param name="name"></param>
+        /// <exception cref="ArgumentNullException"></exception>
+        public static void AssertNotNull<T>(T value, string name)
+        {
+            if (value is null)
+            {
+                throw new ArgumentNullException(name);
+            }
+        }
+
+        /// <summary>
+        /// TBD.
+        /// </summary>
+        /// <param name="value"></param>
+        /// <param name="name"></param>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <exception cref="ArgumentException"></exception>
+        public static void AssertNotNullOrEmpty(string value, string name)
+        {
+            if (value is null)
+            {
+                throw new ArgumentNullException(name);
+            }
+
+            if (value.Length == 0)
+            {
+                throw new ArgumentException("Value cannot be an empty string.", name);
+            }
+        }
+
+        /// <summary>The default message used by <see cref="OperationCanceledException"/>.</summary>
+        private static readonly string s_cancellationMessage = new OperationCanceledException().Message; // use same message as the default ctor
+
+        /// <summary>Throws a cancellation exception if cancellation has been requested via <paramref name="cancellationToken"/>.</summary>
+        /// <param name="cancellationToken">The token to check for a cancellation request.</param>
+        public static void ThrowIfCancellationRequested(CancellationToken cancellationToken)
```

Comment:
I added these which makes this type not make sense to call Argument anymore, @KrzysztofCwalina.  There are more helpers like this in our internal shared source that we might want to put on such a utility library to standardize behavior across clients, but I'm flexible and we can think about alternatives here.